api/v1alpha1/nestedapiserver_types.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2020.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	addonv1alpha1 "sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/apis/v1alpha1"
+)
+
+// NestedAPIServerSpec defines the desired state of NestedAPIServer
+type NestedAPIServerSpec struct {
+	// NestedComponentSpec contains the common and user-specified information that are
+	// required for creating the component
+	// +optional
+	NestedComponentSpec `json:",inline"`
+}
+
+// NestedAPIServerStatus defines the observed state of NestedAPIServer
+type NestedAPIServerStatus struct {
+	// Ready is set if all resources have been created
+	// +kubebuilder:default=false
+	Ready bool `json:"ready,omitempty"`
+
+	// APIServerService is the reference to the service that expose the APIServer
+	// +optional
+	APIServerService *corev1.ObjectReference `json:"apiserverService,omitempty"`
```

Comment:
What do you think about not referencing the actual service but changing the `EctdAddress` type to be a shared `Address` type and we use this feed to represent `Endpoint`.

```suggestion
	// Endpoint is the reference to the address that exposes the APIServer
	// +optional
	Endpoint Address `json:"endpoint,omitempty"`
```

Reply:
Yes. Sounds good to me